R/transactions.R

```diff
@@ -43,4 +57,29 @@ setGeneric("dbCommit",
 setGeneric("dbRollback",
   def = function(conn, ...) standardGeneric("dbRollback"),
   valueClass = "logical"
-)
\ No newline at end of file
+)
+
+#' @aliases withTransaction,DBIConnection-method
+#' @export
+#' @rdname transactions
+setGeneric("withTransaction",
+  def = function(conn, statement, ...) standardGeneric("withTransaction"),
+  valueClass = "logical"
+)
+
+#' @export
+setMethod("withTransaction", "DBIConnection", function(conn, statement, ...) {
```

Comment:
@krlmlr but for that to work, you can't simply rely on forcing the promise, you need to use NSE to capture the expression and re-evaluate multiple times. That doesn't feel like DBI to me.